rust/lance-encoding/src/decoder.rs

```diff
@@ -247,15 +249,23 @@ pub struct ColumnInfo {
     pub page_infos: Vec<Arc<PageInfo>>,
     /// File positions of the column-level buffers
     pub buffer_offsets: Vec<u64>,
+    /// Sizes (in bytes) of the column-level buffers
+    pub buffer_sizes: Vec<u64>,
 }
 
 impl ColumnInfo {
     /// Create a new instance
-    pub fn new(index: u32, page_infos: Vec<Arc<PageInfo>>, buffer_offsets: Vec<u64>) -> Self {
+    pub fn new(
+        index: u32,
+        page_infos: Vec<Arc<PageInfo>>,
+        buffer_offsets: Vec<u64>,
+        buffer_sizes: Vec<u64>,
```

Comment:
I like the idea of adding buffer sizes.  This is helpful information that should be available.

Minor nit:  can we change `buffer_offsets` and `buffer_sizes` to be `Vec<(u64, u64)>` or `Vec<SomeStructWithOffsetAndSize>`?  I'd rather have one `Vec` of pairs than two `Vec`s that are the same size.

Reply:
Sure. I change them to be `Vec<(u64, u64)>`, and also change all related places as well.